ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryPartitionStateModel.java

```diff
@@ -28,10 +28,13 @@
   private Logger logger = LoggerFactory.getLogger(getClass());
   private final String resourceName;
   private final String partitionName;
+  private final PartitionStateChangeListener partitionStateChangeListener;
 
-  AmbryPartitionStateModel(String resourceName, String partitionName) {
+  AmbryPartitionStateModel(String resourceName, String partitionName,
+      PartitionStateChangeListener partitionStateChangeListener) {
     this.resourceName = resourceName;
     this.partitionName = partitionName;
+    this.partitionStateChangeListener = partitionStateChangeListener;
```

Comment:
Our practice is:
this.partitionStateChangeListener Objects.requireNonNull(partitionStateChangeListener, errorMsg);
Which throws a NPE instead of IAE, but we had discussion a while back and agreed on this.